### DISCUSSION TITLE
create_and_notify behavior on invalid new files

### DISCUSSION THREAD

**Discussion by learner-long-life** [nvim-tree/nvim-tree.lua#2330] [Q&A]
Issue:
Confusing error message when creating a new path that assumes some directories that are actually files.

Details:
Ahoy, I was trying to create a new directory with nvim-tree and accidentally created a new file instead. Afterwards, creating a new file in this not-a-directory fails with a confusing error message.

Steps To Reproduce:
* Start `nvim`
* Press `C-n` to open `nvim-tree`
* Press `a` to add a file in current directory, say `math`
* Press `a` again to add another new file, say `math/Blah.scala` (`nvim` doesn't let you auto-complete this, but also doesn't stop you from adding a `/` to any path)
* Receive the following (correct) error

```
5108: Error executing lua: ...y/nvim-tree.lua/lua/nvim-tree/actions/fs/create-file.lua:17: bad argument #1 to 'fs_close' (number expected, got nil)
stack traceback:
        [C]: in function 'fs_close'
        ...y/nvim-tree.lua/lua/nvim-tree/actions/fs/create-file.lua:17: in function 'create_and_notify'
        ...y/nvim-tree.lua/lua/nvim-tree/actions/fs/create-file.lua:79: in function 'on_confirm'
        ...ew/Cellar/neovim/0.9.0/share/nvim/runtime/lua/vim/ui.lua:103: in function 'input'
        ...y/nvim-tree.lua/lua/nvim-tree/actions/fs/create-file.lua:51: in function 'f'
        ...ocal/share/nvim/lazy/nvim-tree.lua/lua/nvim-tree/api.lua:32: in function <...ocal/share/nvim/lazy/nvim-tree.lua/lua/nvim-tree/api.lua:30>
```
 While this behavior is correct, as `math` is not a directory, the error message could be clearer in this `nvim-tree` function
```
local function create_and_notify(file)
  local ok, fd = pcall(vim.loop.fs_open, file, "w", 420)
  if not ok then
    notify.error("Couldn't create file " .. file)
    return
  end
  vim.loop.fs_close(fd)
  events._dispatch_file_created(file)
end
```
For some reason, trying to create a file like this returns `ok` from `pcall` and doesn't return, making it down to the line `vim.loop.fs_close`. Suggestions or improvements? Not a big enough issue?

If you try to create any other path that doesn't exist, it appears to perform the equivalent of `mkdir -p`, that is, creating `./clojure/lalaland/MathUtils.scala` will work beautifully and create the directories `./clojure` and `./clojure/lalaland` automatically.

Thanks.

**Comment by gegoune** [comment]
This definitely isn't valid operation, but error message could be improved. Don't think we can/want to do any validation in user prompt.
